src/supervised/distance.jl

```diff
@@ -112,8 +112,8 @@ It is strictly convex.
 const L2DistLoss = LPDistLoss{2}
 
 value(loss::L2DistLoss, difference::Number) = abs2(difference)
-deriv(loss::L2DistLoss, difference::T) where {T<:Number} = T(2) * difference
-deriv2(loss::L2DistLoss, difference::T) where {T<:Number} = T(2)
+deriv(loss::L2DistLoss, difference::T) where {T<:Number} = convert(T,2) * difference
+deriv2(loss::L2DistLoss, difference::T) where {T<:Number} = convert(T,2)
```

Comment:
the changes from `T(x)` to `convert(T,x)` seem unfortunate, but that seems to be the right thing to do here.

Reply:
Agreed.  Ah, that must be what all the dual number warnings were about...